File: ui/lib/types/system.ts
import type { Settings } from './settings'

export type RuntimeHealth = {
  api_available: boolean
  sse_supported: boolean
  api_version: string
  server_time: string
  runtime_mode: string
}

export type SessionStats = {
  total_runs: number
  queued_runs: number
  running_runs: number
  complete_runs: number
  error_runs: number
  latest_run_id: string | null
}

export type RuntimeConstraints = {
  min_rounds: number
  max_rounds: number
}

export type RuntimeSnapshot = {
  health: RuntimeHealth
  session: SessionStats
  constraints: RuntimeConstraints
  defaults: Settings
}

export type ProviderModels = {
  provider: string
  models: string[]
  error: string | null
}

/** GET /api/symbols/resolve — Yahoo Finance validation before launch */
export type TickerResolveResponse = {
  valid: boolean
  query: string
  yahoo_symbol: string
  display_name: string | null
  message: string | null
}

File: api/models/system.py
from pydantic import BaseModel
from api.models.settings import Settings


class RuntimeHealth(BaseModel):
    api_available: bool
    sse_supported: bool
    api_version: str
    server_time: str
    runtime_mode: str


class SessionStats(BaseModel):
    total_runs: int
    queued_runs: int
    running_runs: int
    complete_runs: int
    error_runs: int
    latest_run_id: str | None = None


class RuntimeConstraints(BaseModel):
    min_rounds: int
    max_rounds: int


class RuntimeSnapshot(BaseModel):
    health: RuntimeHealth
    session: SessionStats
    constraints: RuntimeConstraints
    defaults: Settings


class ProviderModels(BaseModel):
    provider: str
    models: list[str]
    error: str | None = None
